### DISCUSSION TITLE
Perform a search between two dates

### DISCUSSION THREAD

**Discussion by juniortestjr** [andregasser/bigbone#380] [Q&A]
Here is an example code from github : 

```java
import social.bigbone.MastodonClient;
import social.bigbone.api.entity.Search;
import social.bigbone.api.exception.BigBoneRequestException;

public class PerformSimpleSearch {
    public static void main(final String[] args) throws BigBoneRequestException {
        final String instance = args[0];
        final String accessToken = args[1];
        final String searchTerm = args[2];

        // Instantiate client
        final MastodonClient client = new MastodonClient.Builder(instance)
                .accessToken(accessToken)
                .build();

        // Perform search and print results
        final Search searchResult = client.search().searchContent(searchTerm).execute();
        searchResult.getAccounts().forEach(account -> System.out.println(account.getDisplayName()));
        searchResult.getStatuses().forEach(status -> System.out.println(status.getContent()));
        searchResult.getHashtags().forEach(hashtag -> System.out.println(hashtag.getName()));
    }
}
```

And I would like to do a search where the results are between 2 dates. 
But I do not know how. 
Could you help me ?

**Comment by PattaFeuFeu** [comment]
Hi @juniortestjr! Thanks for your question. 😊 

From what I can see in the [API documentation of the search method](https://docs.joinmastodon.org/methods/search/#v2) you’re referring to, it doesn’t look like we _can_ limit the search to _dates_.

We have implemented the search API fully, so we allow you to use all parameters available in the API. There just isn’t any parameter for _dates_ unfortunately.

What you _could_ do is limit by _ids_. For that, you can use the `Range` parameter of `search().searchContent()`. Not sure if that’ll help you though.

What you could also do is get a larger quantity of search results and then filtering by date locally in your application. Not the greatest solution as you’d need to get loads of results first that you might not even be interested in. But I currently see no other way from what the API allows us to do.

**Reply by juniortestjr** [reply]
Thanks for your response. I'll try to see what I can do with what you answered. Thank you so much for your response.
